libsolidity/formal/SymbolicState.cpp

```diff
@@ -114,13 +114,13 @@ void SymbolicState::transfer(smtutil::Expression _from, smtutil::Expression _to,
 {
 	unsigned indexBefore = m_state.index();
 	addBalance(_from, 0 - _value);
-	addBalance(_to, move(_value));
+	addBalance(_to, std::move(_value));
 	unsigned indexAfter = m_state.index();
 	solAssert(indexAfter > indexBefore, "");
 	m_state.newVar();
 	/// Do not apply the transfer operation if _from == _to.
 	auto newState = smtutil::Expression::ite(
-		move(_from) == move(_to),
+		std::move(_from) == std::move(_to),
```

Comment:
This is a weird use of `move()`. It would make sense of `==` took arguments by value - but why would it do that?

Reply:
It seems it's because the parameters are moved into the returned object..
```
	friend Expression operator==(Expression _a, Expression _b)
	{
		smtAssert(_a.sort->kind == _b.sort->kind, "Trying to create an 'equal' expression with different sorts");
		return Expression("=", std::move(_a), std::move(_b), Kind::Bool);
	}
```

Reply:
Hmm... interesting. I guess it's good then.

Reply:
I agree that it seems an unexpected way to use `operator==`